### PR TITLE
feat(tracking): streamline code for every type of consent we have to check [WPB-10506]

### PR DESCRIPTION
**Pull request by aweiss-dev**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10506" title="WPB-10506" target="_blank">TaskWPB-10506</a>  [Web] Streamline the consent settings for Webapp for sharing data
  </td></table>
  <br />
 

## Description

The implementation of how we track and work with the users consent to usage of private data was different. This gets aligned during this PR.

## Checklist

- [ x ] PR has been self reviewed by the author;
- [ x ] Hard-to-understand areas of the code have been commented;
- [ x ] If it is a core feature, unit tests have been added;

### Important details for the reviewers

Note: We still have a environment variable to disable all consent features, that could be removed. I rolled back code to remove it since we have to wait for the other platforms to also remove it.

### CODE REVIEW

File: src/script/team/TeamService.ts [hunk 1]
```diff
@@ -87,6 +87,9 @@ export class TeamService {
         },
         [FEATURE_KEY.CONFERENCE_CALLING]: {
           status: FeatureStatus.ENABLED,
+          config: {
+            useSFTForOneToOneCalls: false,
```
Comment (aweiss-dev) [inline src/script/team/TeamService.ts:91]: This is the change that was needed after the Core update

Comment (V-Gira) [reply src/script/team/TeamService.ts:91]: I think setting it to false in this PR is the correct call, this only affects the release branch at the moment

File: src/script/tracking/EventTrackingRepository.ts [hunk 2]
```diff
@@ -108,12 +96,12 @@ export class EventTrackingRepository {
         this.stopProductReporting();
       }
     } catch (error) {
-      this.logger.warn(`Failed to send new countly tracking id to other devices ${error}`);
+      this.logger.warn(`adrian Failed to send new countly tracking id to other devices ${error}`);
```
Comment (V-Gira) [inline src/script/tracking/EventTrackingRepository.ts]: Hi bud :) 
```suggestion
      this.logger.warn(`Failed to send new countly tracking id to other devices ${error}`);
```

**Review by V-Gira — APPROVED** [review]
LGTM, please do check my comment and remove the leftover log before merging :pray:

File: src/script/tracking/EventTrackingRepository.ts (hunk 2 above)
Comment (aweiss-dev) [reply src/script/tracking/EventTrackingRepository.ts]: Good catch :D

File: src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx [hunk 3]
```diff
@@ -43,41 +40,37 @@ interface DataUsageSectionProps {
   teamState?: TeamState;
 }
 
-const DataUsageSection: React.FC<DataUsageSectionProps> = ({
-  propertiesRepository,
-  brandName,
-  isActivatedAccount,
-  teamState = container.resolve(TeamState),
-}) => {
+const DataUsageSection: React.FC<DataUsageSectionProps> = ({propertiesRepository, brandName, isActivatedAccount}) => {
```
Comment (przemvs) [inline src/script/page/MainContent/panels/preferences/accountPreferences/DataUsageSection.tsx]: We trying to not to use React.FC :)
```suggestion
const DataUsageSection = ({propertiesRepository, brandName, isActivatedAccount}: DataUsageSectionProps) => {
```

**Pull request merged**